apps/angular/src/app/features/anime/anime-details/anime-details.component.ts

```diff
@@ -0,0 +1,75 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AnimeService } from '@js-camp/angular/core/services/anime.service';
+import { AnimeDetails } from '@js-camp/core/models/anime-details';
+import { EMPTY, Observable, catchError, switchMap } from 'rxjs';
+
+import { ImageDialogComponent } from './image-dialog/image-dialog.component';
+
+/** Anime details component. */
+@Component({
+	selector: 'camp-anime-details',
+	templateUrl: './anime-details.component.html',
+	styleUrls: ['./anime-details.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AnimeDetailsComponent {
+	/** Image popup state. */
+	protected isPopupOpened = false;
+
+	/** Response observable. */
+	protected readonly animeDetails$: Observable<AnimeDetails>;
+
+	public constructor(
+		private readonly activatedRoute: ActivatedRoute,
+		private readonly router: Router,
+		private readonly animeService: AnimeService,
+		public readonly dialog: MatDialog,
```

Comment:
There is no need to make this property 'public'. We can make it `private`:
```suggestion
		private readonly dialog: MatDialog,
```